src/Classes/ListingHydrator.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace ListingsApp\Classes;
+use PDO;
+
+/**
+ * Class ListingHydrator
+ * namespace - ListingsApp\Classes
+ */
+class ListingHydrator
+{
+    /**
+     * FETCH_PROPS_LATE - hydrates Listing class AFTER constructor has run
+     *
+     * @param PDO $db
+     * @param string $agentRef - string delivered to listing.php via GET request
+     * @return Listing - a fully hydrated Listing object
+     */
+    public function getListing(PDO $db, string $agentRef): Listing
+    {
+        $query= $db->prepare('SELECT `agent_ref`, `address_1`,  `address_2`, `town`, `postcode`,  `description`, `bedrooms`, `price`, `image`, `type`, `status_name` AS `status` FROM `listings` INNER JOIN `statuses` ON `listings`.`status` = `statuses`.`id` WHERE `agent_ref` = :agentRef;');
+        $query->bindParam('agentRef', $agentRef);
+        $query->setFetchMode(PDO::FETCH_CLASS | PDO::FETCH_PROPS_LATE, Listing::class);
+        $query->execute();
+        return $query->fetch();
+    }
+
+    /**
+     * gets all listings from db table listings
+     * @param PDO $db
+     * @return array
+     */
+    public static function getAllListings(PDO $db): array
+    {
+        $query = $db->prepare("SELECT `agent_ref`, `address_1`, `address_2`, `town`, `postcode`, `description`, `bedrooms`, `price`, `image`, `type`, `status` FROM `listings`;");
```

Comment:
This query needs to be updated to join in the statuses table (in the same way as the above query)